python/gps/gps_main.py

```diff
@@ -352,10 +355,19 @@ def _log_data(self, itr, traj_sample_lists, pol_sample_lists=None):
             pol_sample_lists: policy samples as SampleList object
         Returns: None
         """
+
+        ## If IOC
+        if self.using_ioc():
+            demo_losses = eval_demos_xu(self.agent, self.algorithm.demoX, self.algorithm.demoU, self.algorithm.cost, n=5)
+            sample_losses = eval_demos_xu(self.agent, traj_sample_lists[0].get_X(), traj_sample_lists[0].get_U(), self.algorithm.cost, n=5)
```

Comment:
I think you should be able to get this from self.algorithm.cur[0].cs (for condition 0). Right now, it is pretty cheap to just re-evaluate, but it will be slower when vision is involved.